src/algokit/core/tasks/mint/mint.py

```diff
@@ -0,0 +1,154 @@
+
+
+def _file_mimetype(filename: pathlib.Path) -> str:
+    extension = pathlib.Path(filename).suffix
+    return mimetypes.types_map[extension]
+
+
+def _create_asset_txn(
+    *,
+    asset_config_params: AssetConfigTxnParams,
+    token_metadata: TokenMetadata,
+    use_metadata_hash: bool = True,
+) -> transaction.AssetConfigTxn:
+    json_metadata = token_metadata.to_json()
+    metadata = json.loads(json_metadata)
+
+    if use_metadata_hash:
+        if "extra_metadata" in metadata:
+            h = hashlib.new("sha512_256")
+            h.update(b"arc0003/amj")
+            h.update(json_metadata.encode("utf-8"))
+            json_metadata_hash = h.digest()
+
+            h = hashlib.new("sha512_256")
+            h.update(b"arc0003/am")
+
+            h.update(json_metadata_hash)
+            h.update(base64.b64decode(metadata["extra_metadata"]))
+            asset_config_params.metadata_hash = h.digest()
+        else:
+            h = hashlib.new("sha256")
+            h.update(json_metadata.encode("utf-8"))
+            asset_config_params.metadata_hash = h.digest()
+    else:
+        asset_config_params.metadata_hash = b""
+
+    return transaction.AssetConfigTxn(**asdict(asset_config_params))  # type: ignore[no-untyped-call]
+
+
+def mint_token(  # noqa: PLR0913
+    *,
+    client: algod.AlgodClient,
+    api_key: str,
+    creator_account: Account,
+    asset_name: str,
+    unit_name: str,
+    total: int,
+    token_metadata: TokenMetadata,
+    metadata_standard: str,
+    is_mutable: bool,
+    image_path: pathlib.Path | None = None,
+    decimals: int | None = 0,
+) -> tuple[int, str]:
+    if metadata_standard not in [MetadataStandard.ARC19.value, MetadataStandard.ARC3.value]:
+        raise ValueError("Invalid metadata standard provided!")
+
+    if not token_metadata.name or token_metadata.name != asset_name:
+        raise ValueError("Token name in metadata JSON must match CLI argument providing token name!")
+
+    if token_metadata.decimals and token_metadata.decimals != decimals:
+        raise ValueError("Token metadata JSON and CLI arguments providing decimals amount must be equal!")
+
+    if image_path:
+        token_metadata.image_integrity = _file_integrity(image_path)
+        token_metadata.image_mimetype = _file_mimetype(image_path)
+        logger.info("Uploading image to Web3 Storage...")
+        token_metadata.image = "ipfs://" + upload_to_web3_storage(image_path, api_key=api_key)
+        logger.info(f"Image uploaded to Web3 Storage: {token_metadata.image}")
+
+    logger.info("Uploading metadata to Web3 Storage...")
+    metadata_cid = upload_to_web3_storage(
+        token_metadata.to_file_path(),
+        api_key=api_key,
+    )
+    logger.info(f"Metadata uploaded to Web3 Storage: {metadata_cid}")
+
+    asset_config_params = AssetConfigTxnParams(
+        sender=creator_account.address,
+        sp=client.suggested_params(),
+        reserve=_reserve_address_from_cid(metadata_cid),
```

Comment:
This should only happen for ARC-19?

Reply:
@robdmoore good catch, should i use creator_address by default for arc3 then or keep it None?

Reply:
@robdmoore setting it to None for case of arc3